string mode
string aux_info
float32 data1
float32 data2
float32 data3